websites/D/Discords.com/presence.ts

```diff
@@ -3,351 +3,518 @@ const presence = new Presence({
+			},
+			"/apply": {
+				details: "📝 Applying for: Verification",
+				buttons: [{ label: "Apply", url: href }],
+			},
+		},
+		pageNumber = document.querySelector("li.page-item.active")?.textContent,
+		pageNumberBots = document.querySelector("li.active")?.textContent,
+		tagSearch = document
+			.querySelector("h1.text-md-left")
+			?.textContent.replace("Discord Servers", " "),
+		standardSearch = document
+			.querySelector("div.FMDiWzVk")
+			?.querySelector("input")
+			?.getAttribute("value");
+
+	for (const [path, data] of Object.entries(pages))
+		if (pathname.includes(path)) presenceData = { ...presenceData, ...data };
 
-	if (document.location.pathname === "/")
+	if (pathname === "/") {
 		presenceData.details = "Viewing home page";
-	else if (document.location.pathname === "/servers/search") {
+
+		if (standardSearch) {
+			presenceData.details = `🔍 Searching for: ${standardSearch}`;
+			presenceData.buttons = [
+				{
+					label: "View Results",
+					url: href,
+				},
+			];
+		}
+	} else if (pathname === "/servers/search") {
 		presenceData.details = `🔍 Searching for: ${
 			document
 				.querySelector("h1.text-center")
 				?.textContent.replace("Discord Servers", " ") || "Nothing"
 		}`;
-		presenceData.state = `📖 Page ${
-			document.querySelector("li.page-item.active")?.textContent
-		}`;
+		presenceData.state = `📖 Page ${pageNumber}`;
 		presenceData.buttons = [
 			{
 				label: "View Results",
-				url: document.location.href,
+				url: href,
 			},
 		];
-	} else if (document.location.pathname.includes("/servers/tag/")) {
-		presenceData.details = `Viewing 📛 ${
-			document
-				.querySelector("h1.text-center")
-				?.textContent.replace("Discord Servers", " ") || "Nothing"
-		} tag`;
-		presenceData.state = `📖 Page ${
-			document.querySelector("li.page-item.active")?.textContent
-		}`;
+	} else if (pathname.includes("/servers/tag/")) {
+		if (tagSearch) presenceData.details = `🏷️ Viewing tag: ${tagSearch}`;
+		presenceData.state = `📖 Page ${pageNumber}`;
 		presenceData.buttons = [
 			{
 				label: "View Tag",
-				url: document.location.href,
+				url: href,
 			},
 		];
-	} else if (document.location.pathname.includes("/servers/tags/")) {
-		presenceData.details = `Viewing 🔑 ${
+	} else if (pathname.includes("/servers/tags/")) {
+		if (tagSearch) presenceData.details = `🔑 Viewing keyword: ${tagSearch}`;
+		presenceData.state = `📖 Page ${pageNumber}`;
+		presenceData.buttons = [
+			{
+				label: "View Keyword",
+				url: href,
+			},
+		];
+	} else if (pathname === "/servers/top-100") {
+		presenceData.details = "🏅 Viewing: top-100 servers";
+		presenceData.state = `📖 Page ${pageNumber}`;
+		presenceData.buttons = [
+			{
+				label: "View Top-100",
+				url: href,
+			},
+		];
+	} else if (pathname.includes("/join")) {
+		presenceData.details = "🔗 Joining";
+		presenceData.state = document
+			.querySelector("h2.__9ui9XW8A.mt-4")
+			?.textContent.slice(0, 128);
```

Comment:
This functionality is already built in to the extension.
```suggestion
			?.textContent;
```

Reply:
Oh. So remove .slice everywhere?

Reply:
Yeah, it's not needed for `details` and `state`